### DISCUSSION TITLE
How to import raw json?

### DISCUSSION THREAD

**Discussion by onnyyonn** [ImranR98/Obtainium#1739] [Q&A]
When I add an app from https://apps.obtainium.imranr.dev by clicking on "Add to Obtainium", Obtainium shows the raw json that would be imported. If I have this json for an app, is there any way to import it manually? I looked into the "Import/Export" tab, and couldn't find anything similar. "Obtainium Import" fails to import these.

**Comment by DwainZwerg** [comment]
You can simply move the json file to Android\data\dev.imranr.obtainium\data. If it is correct, the app will be in the app list the next time you start it.

**Reply by onnyyonn** [reply]
Thanks. But this directory is not accessible without root, is it?

**Reply by DwainZwerg** [reply]
Indeed, this has nothing to do with root. ~\Android\data is located in the user directory (path \storage\emulated\0\Android\data\), where all other things such as documents (\storage\emulated\0\documents) etc. are also located. Recently (Android 13+), Google has come up with something new to bully third-party file managers (they justify it with security improvements, which this approach ultimately is, because a system always becomes more secure when fewer apps are allowed to do fewer things). The only file manager that can reliably access \storage\emulated\0\Android\ is the rudimentary file manager built into Android by default. However, because it is not displayed or hidden in most launchers (the app that displays the home screen and app list), there is [this app](https://play.google.com/store/apps/details?id=com.marc.files) in the Play Store, which creates a link to the system file manager.

**Reply by no-usernames-left** [reply]
> You can simply move the json file to Android\data\dev.imranr.obtainium\data. If it is correct, the app will be in the app list the next time you start it.

This doesn't solve the issue, as the website copies it to the clipboard instead of offering a file.

**Comment by callmeTayyy** [comment]
@onnyyonn there's 2 types of json Obtainium accepts:

The HTML links you click on, like the ones on the website you mentioned. You can save those as Markdown or HTML to use later.

To export:
1. select the app you want to export
2. click on the three dots menu on the bottom of the screen
3. click on 'Share app configuration as HTML link'
4. save it as Markdown, HTML or other format you click the link

---

There's also the json Obtainium exports, those you can save as '.json' and import later.

To export:
1. select the app you want to export
2. click on the three dots menu on the bottom of the screen
3. click on 'Share - Obtainium export' (figure it out a way to save it)

To import:
1. go to 'Import/export'
2. click on 'Obtainium import'
3. select your '.json' file
